packages/webview_flutter/webview_flutter_platform_interface/lib/src/method_channel/webview_method_channel.dart

```diff
@@ -84,7 +84,31 @@ class MethodChannelWebViewPlatform implements WebViewPlatformController {
   @override
   Future<void> loadFile(String absoluteFilePath) async {
     assert(absoluteFilePath != null);
-    return _channel.invokeMethod<void>('loadFile', absoluteFilePath);
+
+    try {
+      return await _channel.invokeMethod<void>('loadFile', absoluteFilePath);
+    } on PlatformException catch (ex) {
+      if (ex.code == 'loadFile_failed') {
+        throw ArgumentError(ex.message);
+      }
+
+      rethrow;
+    }
+  }
+
+  @override
+  Future<void> loadFlutterAsset(String key) async {
+    assert(key.isNotEmpty);
+
+    try {
+      return await _channel.invokeMethod<void>('loadFlutterAsset', key);
+    } on PlatformException catch (ex) {
+      if (ex.code == 'loadFlutterAsset_failed') {
```

Comment:
Since we don't have to be compatible with an existing implementation (unlike loadFile) let's give this a specific code like `loadFlutterAsset_invalidKey`. If we ever have failure cases that aren't due to a bad key for some reason, we would't want them mapped to `ArgumentError`.